Introduce initial Rome website setup

weblancaster opened the pull request:
Introduces the very minimum to get a website up and running.
Includes: `Getting Started`, `Code of Conduct` and `Contributing`

- No custom design.
- Uses the temporary logo #1 .
- The feature list is randomly taken from README.

Assumes the website is part of the repository but can (should?) be extracted
to a separate repository.

Generated using Facebook template documented in [Docusaurus](https://v2.docusaurus.io/docs/installation#scaffold-project-website)

Addresses #14 

![Screen Shot 2020-02-29 at 2 18 38 PM](https://user-images.githubusercontent.com/549394/75612018-8226df00-5afe-11ea-96b5-e9e0d7c754b1.png)

Daniel15 commented on website/.eslintcache:
```diff
@@ -0,0 +1 @@
+[{"/Users/yangshun/Developer/docusaurus_users/docusaurus/packages/docusaurus-init/templates/facebook/docusaurus.config.js":"1","/Users/yangshun/Developer/docusaurus_users/docusaurus/packages/docusaurus-init/templates/facebook/sidebars.js":"2","/Users/yangshun/Developer/docusaurus_users/docusaurus/packages/docusaurus-init/templates/facebook/src/pages/index.js":"3"},{"size":2777,"mtime":15776092409 ...
```
Is this file needed? @yangshun Why is `/Users/yangshun` here? 🤔

weblancaster replied:
Good question, I figured it came from the FB template but I will remove it. thanks.

yangshun replied:
I forgot to ignore the `.eslintcache` file during publishing. My bad, this is an error upstream!

Daniel15 commented on website/docusaurus.config.js:
```diff
@@ -0,0 +1,98 @@
+        docs: {
+          sidebarPath: require.resolve('./sidebars.js'),
+          editUrl:
+            'https://github.com/facebook/docusaurus/edit/master/website/',
```
This needs to be adjusted.

weblancaster replied:
Yep, I can remove it.

sebmck commented:
Thanks for getting this started @weblancaster! I'll try to review this when I can.

sebmck commented:
@Daniel15 

> I wonder if the site should live in `/packages/@romejs/website` or something like that. @sebmck What do you think?

I'd prefer to keep the `packages` directory with only source code. Happy with the `website` folder approach.

yangshun commented:
> I'd prefer to keep the `packages` directory with only source code. Happy with the `website` folder approach.

I agree. `packages` should contain stuff to be published onto a registry, and the `website` won't be published. See https://github.com/facebook/docusaurus repo's structure.

yangshun commented on website/docs/community/code-of-conduct.md:
```diff
@@ -0,0 +1,51 @@
+---
```
This file probably doesn't need to be in the website. We can link the website to the corresponding file in the repo.

weblancaster replied:
Makes sense, I will make the change. thanks

yangshun commented on website/package.json:
```diff
@@ -0,0 +1,46 @@
+{
+  "name": "website",
```
How about calling it `rome-website`?

weblancaster replied:
Makes sense, I will make the change. thanks

yangshun commented on website/docusaurus.config.js:
```diff
@@ -0,0 +1,96 @@
+          title: 'More Resources',
+          items: [
+            {
+              label: 'Github',
```
Nit - GitHub (capital H)

weblancaster replied:
Good catch. I will make the change. thanks

weblancaster commented:
@yangshun I addressed your comments. Now, regarding the color theme, although I made the change based on the logo, it's going to change so not worth the effort at the moment.

sebmck commented:
Happy to defer review to @yangshun or @Daniel15! I don't have strong opinions or any familarity with docusaurus so I'm good with anything.

yangshun commented:
LGTM but `package.json` has a conflict now. @sebmck if you don't mind I can add myself to the repo and help review future website PRs/diffs.

sebmck commented:
Sure! That would be amazing!

sebmck commented:
Thank you so much @weblancaster and @Daniel15/@yangshun for the review! I just opened #73 that removes some of the deps added.

yangshun commented:
I'll try to come up with a customized landing page design for the website sometime within the next few days!

Daniel15 commented:
I can configure the site on Netlify once an admin of the "facebookexperimental" org approves the Netlify GitHub app for use in the org (maybe @JoelMarcey or @zpao can do that). We use Netlify for a few other open-source project sites (Yarn, React, Babel, etc) and for open-source projects like this I've got access to a customized plan we can use :)

yangshun commented:
Do you mean for the live site? We can use gh-pages too. But need them to
enable it still.

On Sun, Mar 1, 2020, 3:59 PM Daniel Lo Nigro <notifications@github.com>
wrote:

> I can configure the site on Netlify once an admin of the
> "facebookexperimental" org approves the Netlify GitHub app for use in the
> org (maybe @JoelMarcey <https://github.com/JoelMarcey> or @zpao
> <https://github.com/zpao> can do that). We use Netlify for a few other
> open-source project sites (Yarn, React, Babel, etc) and for open-source
> projects like this I've got access to a customized plan we can use :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/facebookexperimental/rome/pull/62?email_source=notifications&email_token=AAKBCHLEBIAA3TI72QQXF4LRFIIWFA5CNFSM4K5J4VGKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENMYAMA#issuecomment-593068080>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAKBCHIWBJW3B55SQMJBFN3RFIIWFANCNFSM4K5J4VGA>
> .
>

Daniel15 commented:
Yeah. Netlify is easier to configure and use (you don't have to commit the generated HTML files to source control), more flexible and has more features, but either is fine.

sebmck commented:
I own the domains `romejs.dev, `romejs.io`, `romejs.com `, and `romejs.org`. They’re currently all setup to redirect to `romejs.dev`. Let me know what DNS records I need to add for Netlify.

JoelMarcey commented:
This is exciting! 😀

@Daniel15, @yangshun 

I enabled Netlify for the repo. If we use it, we should add it under our Facebook Open Source Netlify account to keep it consolidated with our other sites that use Netlify for building.

Generally, for our open source projects, Netlify is used for building and previewing the site builds, with GitHub Pages serving the live site. But it could work Netlify doing the serving.

> I own the domains romejs.dev, romejs.io, romejs.com , and romejs.org. They’re currently all setup to redirect to romejs.dev.

@sebmck At some point, we will probably want to transfer the domains over to our internal domain management (like we have for our other open source domains) and then use our CDN (Cloudflare) to manage the DNS records.

Daniel15 commented:
Thank you Joel! I can set up Netlify today. 

> If we use it, we should add it under our Facebook Open Source Netlify account to keep it consolidated with our other sites that use Netlify for building.

@JoelMarcey are you able to add me to the FB Netlify team? I don't have access to that one. 

> Generally, for our open source projects, Netlify is used for building and previewing the site builds, with GitHub Pages serving the live site. But it could work Netlify doing the serving.

If you only use Netlify for previews and not the real site, all the features that Netlify have but Github Pages doesn't have will work in PR previews but fail on the live site, which is very confusing :) one basic example I can think of is redirects. Netlify lets you create a `_redirects` file describing redirects, but Github Pages doesn't understand that.

> then use our CDN (Cloudflare) to manage the DNS records.

FWIW Netlify recommend people use their DNS service rather than Cloudflare. If you do use Cloudflare with Netlify, you need to ensure it's not proxied.

JoelMarcey commented:
> are you able to add me to the FB Netlify team? I don't have access to that one.

I added you as a collaborator to https://app.netlify.com/teams/facebook-open-source/ - if you can add the rome site to that, that would be 👍

> If you only use Netlify for previews and not the real site, all the features that Netlify have but Github Pages doesn't have will work in PR previews but fail on the live site, which is very confusing :) one basic example I can think of is redirects. 

Cloudflare lets you handle a bunch of things that GitHub Pages doesn't -- including redirects. 

I want to avoid, if possible, having multiple infrastructure paths for our open source projects. It took me a lot of campaigning and churn to get agreement on what we use now. If we find that there is a show-stopping need that cannot be accomplished by our current recommended and approved setup, then we can revisit - either by seeing how we can fix it with our current setup or move to something else.

That said - we do have a relationship with Netlify, so we could look at doing something deeper with them, but that would take a bit of time and budgeting, and another round of convincing internally.

Daniel15 commented:
> if you can add the rome site to that, that would be 👍

Will do! I set it up under a new "Rome" team but should be able to move it into the FB one. Otherwise I'll delete the site and create a new one on the FB team.

> Cloudflare lets you handle a bunch of things that GitHub Pages doesn't -- including redirects.

You can't configure that through a PR though, so contributors can't add redirects themselves without asking an admin.

Also Cloudflare redirects wouldn't be compatible with Netlify, as Netlify strongly suggest not proxying to their service as it breaks some things and makes the site slower (see https://community.netlify.com/t/common-issue-why-isn-t-my-ssl-certificate-provisioning-automatically-with-cloudflare-netlify-are-there-other-problems-with-using-cloudflare-in-front-of-netlify/138). You can use Cloudflare DNS with Netlify but the site should not be proxied through Cloudflare (proxying should be disabled on the Cloudflare side)

Daniel15 commented:
@JoelMarcey  The site is now configured the way you like - Hosted in the Facebook open-source Netlify account, with DNS hosted by Cloudflare. Thanks for the assistance :)

JoelMarcey commented:
@Daniel15 Awesome. Thanks! Could you ping me internally and we can talk about making sure we are consistent on which accounts things are being setup at.
